File: benchmarks/fixtures/t7-py-architecture/repo/src/permits/adapters/system_clock.py
from __future__ import annotations

import time
from datetime import datetime, timezone


class SystemClock:
    def now_iso(self) -> str:
        return datetime.now(timezone.utc).isoformat(timespec="seconds")

    def monotonic(self) -> float:
        return time.monotonic()


class FrozenClock:
    """Used by the tests. Never wired by :mod:`permits.wiring`."""

    def __init__(self, instant: str, ticks: float = 0.0) -> None:
        self.instant = instant
        self.ticks = ticks

    def now_iso(self) -> str:
        return self.instant

    def monotonic(self) -> float:
        return self.ticks

    def advance(self, seconds: float) -> None:
        self.ticks += seconds

File: benchmarks/fixtures/t7-ts-architecture/repo/src/adapters/systemClock.ts
import type { Clock } from "../ports/clock.ts";

export class SystemClock implements Clock {
  now(): Date {
    return new Date();
  }
  ticks(): number {
    return Date.now();
  }
}

/** Used by the tests; never wired by the container. */
export class FrozenClock implements Clock {
  constructor(private instant: Date) {}
  now(): Date {
    return new Date(this.instant.getTime());
  }
  ticks(): number {
    return this.instant.getTime();
  }
  advance(ms: number): void {
    this.instant = new Date(this.instant.getTime() + ms);
  }
}
